src/NoteMouseHandler/NoteMouseHandler.js

```diff
@@ -0,0 +1,97 @@
+function findMap(arr, func) {
+  for (const item of arr) {
+    const result = func(item)
+    if (result) return result
+  }
+  return null
+}
+
+export default class MouseHandler {
+  constructor(emitter, actionFactories, cursorHandler) {
+    this.emitter = emitter
+    this.actionFactories = actionFactories
+    this.cursorHandler = cursorHandler
+  }
+
+  onMouseDown(e, ctx) {
+    this.action = findMap(this.actionFactories, factory =>
+      factory(e, ctx)
+    )
+    if (!this.action) {
+      return
+    }
+    let actionMouseDown = () => {}
+    this.actionMouseMove = () => {}
```

Comment:
__[jshint]__
*  'arrow function syntax (=>)' is only available in ES6 (use 'esversion

<!-- Comment by [SideCI](https://sideci.com) -->